metadata/metadata_test.go

```diff
@@ -555,6 +555,56 @@ var _ = Describe("Metadata", func() {
 		})
 	})
 
+	When("SetAnnotation is called", func() {
+		It("should add the annotation to the object", func() {
+			key := "foo"
+			value := "bar"
+			pod := &corev1.Pod{}
+
+			Expect(pod.Annotations).To(HaveLen(0))
+			Expect(SetAnnotation(pod, key, value)).To(Succeed())
+			Expect(pod.Annotations).To(HaveKeyWithValue(key, value))
+		})
+
+		It("should add the annotation to the existing ones", func() {
+			key := "foo"
+			value := "bar"
+			pod := &corev1.Pod{
+				ObjectMeta: v1.ObjectMeta{
+					Annotations: map[string]string{"quux": "corge"},
+				},
+			}
+
+			Expect(pod.Annotations).To(HaveLen(1))
+			Expect(SetAnnotation(pod, key, value)).To(Succeed())
+			Expect(pod.Annotations).To(HaveLen(2))
+			Expect(pod.Annotations).To(HaveKeyWithValue(key, value))
+
+		})
+
+		It("should update the existing annotation", func() {
+			key := "foo"
+			value := "bar"
+			pod := &corev1.Pod{
+				ObjectMeta: v1.ObjectMeta{
+					Annotations: map[string]string{key: "corge"},
+				},
+			}
+
+			Expect(pod.Annotations).To(HaveLen(1))
+			Expect(SetAnnotation(pod, key, value)).To(Succeed())
+			Expect(pod.Annotations).To(HaveLen(1))
+			Expect(pod.Annotations).To(HaveKeyWithValue(key, value))
+
+		})
+
+		It("should error if the object is nil", func() {
```

Comment:
Could you move this to the beginning? It's nice when tests follow code order.

Reply:
I copy pasted order of AddAnnotation tests, I can update it, no problem, JFYI

Reply:
Good catch. We can update that in a different PR ;)